File: src/main/java/com/example/demo/model/Notification.java
package com.example.demo.model;

import com.example.demo.model.enums.NotificationType;
import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private NotificationType notificationType;

    private Long elementId;
    private String text;
    private LocalDateTime timestamp;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = true)
    private Users user;
}

File: frontend/licenta-app/src/app/models/notification.ts
import {User} from "./user";

export class Notification {
  id?: number;
  notificationType?: string;
  elementId?: number;
  text?: string;
  timestamp?: string;
  sender?: User;
  receiver?: {
    id: number;
    name: string;
    image?: string;
  };
}
